modules/schema/src/main/java/org/apache/ignite/internal/schema/marshaller/reflection/ObjectStatistics.java

```diff
@@ -19,33 +19,51 @@
 
 import static org.apache.ignite.internal.schema.marshaller.MarshallerUtil.getValueSize;
 
+import java.math.BigDecimal;
 import java.util.List;
+import org.apache.ignite.internal.binarytuple.BinaryTupleCommon;
 import org.apache.ignite.internal.marshaller.Marshaller;
 import org.apache.ignite.internal.schema.Column;
 import org.apache.ignite.internal.schema.SchemaDescriptor;
 import org.apache.ignite.internal.schema.row.RowAssembler;
 import org.apache.ignite.internal.type.NativeType;
+import org.apache.ignite.internal.type.NativeTypeSpec;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * Object statistic.
  */
 class ObjectStatistics {
     /** Cached zero statistics. */
-    private static final ObjectStatistics ZERO_STATISTICS = new ObjectStatistics(-1);
+    private static final ObjectStatistics ZERO_STATISTICS = new ObjectStatistics(-1, false);
+
+    /**
+     * The average size of a decimal number is taken from the air. This value is used to avoid unnecessary
+     * buffer reallocation when building a binary row. The exact size of the decimal value can be determined
+     * after compaction (see {@link BinaryTupleCommon#shrinkDecimal(BigDecimal, int)}.
+     */
+    private static final int DECIMAL_VALUE_SIZE = 32;
```

Comment:
```suggestion
    /**
     * The value is used to avoid unnecessary
     * buffer reallocation when building a binary row. The exact size of the decimal value can be determined
     * after compaction (see {@link BinaryTupleCommon#shrinkDecimal(BigDecimal, int)}.
     */
    private static final int DECIMAL_VALUE_ESTIMATED_SIZE = 32;
```

Reply:
Done, thanks